crates/xhr/src/lib.rs

```diff
@@ -0,0 +1,320 @@
+#![deny(missing_docs, missing_debug_implementations)]
+
+/*!
+  XmlHttpRequest.
+*/
+
+/// The raw XmlHttpRequest API, wrapped with convenient Rust types.
+///
+/// Synchronous requests are not supported (because deprecated).
+pub mod raw {
+    use wasm_bindgen::{closure::Closure, prelude::*, JsCast};
+    use web_sys;
+
+    /// See MDN docs.
+    #[derive(Debug, Clone)]
+    pub struct XmlHttpRequest {
+        xhr: web_sys::XmlHttpRequest,
+    }
+
+    impl XmlHttpRequest {
+        /// Initialize an XmlHttpRequest.
+        pub fn new() -> Self {
```

Comment:
should we assume this is safe because all browsers that support webassembly implement XmlHttpRequest, or check whether the constructor is defined?

Reply:
If an API predates WebAssembly, then yes we can safely assume it exists.

In addition, [XMLHttpRequest goes all the way back to IE7](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest#Browser_compatibility)! So we can safely assume it exists regardless.

Personally, I would just use `unwrap_throw`, but I don't have strong feelings about it.